rpc/wrtc_rate_limiter.go

```diff
@@ -0,0 +1,152 @@
+package rpc
+
+import (
+	"context"
+	"time"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"go.viam.com/utils"
+	mongoutils "go.viam.com/utils/mongo"
+	"go.viam.com/utils/perf/statz"
+	"go.viam.com/utils/perf/statz/units"
+)
+
+// A WebRTCRateLimiter rate limits requests used in WebRTC signaling to control the rate of requests
+// made based on a key, typically a combination of method and auth ID.
+type WebRTCRateLimiter interface {
+	Allow(ctx context.Context, key string) error
+}
+
+func init() {
+	mongoutils.MustRegisterNamespace(&mongodbWebRTCCallQueueDBName, &mongodbRateLimiterCollName)
+}
+
+var rateLimitDenials = statz.NewCounter1[string]("signaling/rate_limits_denials", statz.MetricConfig{
+	Description: "Total number of requests rate limited.",
+	Unit:        units.Dimensionless,
+	Labels: []statz.Label{
+		{Name: "key", Description: "Method and auth ID of the client being rate limited."},
+	},
+})
+
+// Database configuration and collection names for MongoDB rate limiter.
+var (
+	mongodbRateLimiterCollName     = "rate_limiter"
+	mongodbWebRCRateLimiterTTLName = "rate_limit_expire"
+)
+
+type rateLimitDocument struct {
+	ID        primitive.ObjectID `bson:"_id,omitempty"`
```

Comment:
If you're not doing anything with the auto-generated `_id`, I don't think you need to encode it in the document.